lib/api/wasmedge.cpp

```diff
@@ -498,6 +498,51 @@ WASMEDGE_CAPI_EXPORT void WasmEdge_LogSetDebugLevel(void) {
 
 WASMEDGE_CAPI_EXPORT void WasmEdge_LogOff(void) { WasmEdge::Log::setLogOff(); }
 
+WASMEDGE_CAPI_EXPORT void WasmEdge_LogSetLevel(WasmEdge_LogLevel Level) {
+  switch (Level) {
+  case WasmEdge_LogLevel_Trace:
+    WasmEdge::Log::setTraceLoggingLevel();
+    break;
+  case WasmEdge_LogLevel_Debug:
+    WasmEdge::Log::setDebugLoggingLevel();
+    break;
+  case WasmEdge_LogLevel_Info:
+    WasmEdge::Log::setInfoLoggingLevel();
+    break;
+  case WasmEdge_LogLevel_Warn:
+    WasmEdge::Log::setWarnLoggingLevel();
+    break;
+  case WasmEdge_LogLevel_Error:
+    WasmEdge::Log::setErrorLoggingLevel();
+    break;
+  case WasmEdge_LogLevel_Critical:
+    WasmEdge::Log::setCriticalLoggingLevel();
+    break;
+  default:
+    assumingUnreachable();
```

Comment:
This `Level` is provided by the user, so it cannot be assumed to always be valid.
You should check for invalid input here and decide whether to return an error or ignore it.